File: Edward Poot/src/QL/ast/expr/div.go
package expr

import "ql/ast/visit"

type Div struct {
	Lhs, Rhs Expr
}

func (d Div) GetLhs() Expr {
	return d.Lhs
}

func (d Div) GetRhs() Expr {
	return d.Rhs
}

func (d Div) Eval() interface{} {
	return d.GetLhs().Eval().(int) / d.GetRhs().Eval().(int)
}

func (d Div) Accept(v visit.Visitor) interface{} {
	return v.Visit(d)
}

File: Kevin van den Bekerom/DSLQL/src/main/java/org/uva/sea/ql/ast/expr/math/Div.java
package org.uva.sea.ql.ast.expr.math;

import org.uva.sea.ql.ast.expr.BinaryExpr;
import org.uva.sea.ql.ast.expr.Expr;
import org.uva.sea.ql.ast.form.Context;
import org.uva.sea.ql.ast.form.ValueMap;
import org.uva.sea.ql.ast.visit.Visitor;
import org.uva.sea.ql.type.NumericalType;
import org.uva.sea.ql.type.Type;
import org.uva.sea.ql.value.Value;

public class Div extends BinaryExpr {
	
	public Div(Expr lhs, Expr rhs, int startLine) {
		super(lhs, rhs, startLine);
	}
	
	@Override
	public Value eval(ValueMap valueMap) {
		return rhs.eval(valueMap).div(lhs.eval(valueMap));
	}
	
	public void accept(Visitor visitor, Object context) {
		visitor.visit(this, context);
	}

	@Override
	public String toString() {
		return super.toString() + " (/)";
	}
	
	@Override
	public Type getType(Context context) {
		return new NumericalType();
	}
}